src/Guzzle6Promise.php

```diff
@@ -63,7 +63,7 @@ public function __construct(PromiseInterface $promise, RequestInterface $request
             }
 
             if (!($reason instanceof GuzzleExceptions\GuzzleException)) {
-                throw new \RuntimeException("Invalid reason");
+                throw new \RuntimeException("Invalid exception returned from Guzzle6", $reason);
```

Comment:
ups, good point. i will add a type check there. if its not an exception, i will keep ignoring it, we don't consider returning a string as reason a good idea.